static/js/tools/map/chart_store.ts

```diff
@@ -0,0 +1,199 @@
+  };
+  breadcrumbDenomStat: {
+    data: EntitySeriesWrapper;
+    context?: DataContext;
+  };
+  mapPointStat: {
+    data: EntityObservationWrapper;
+    context?: DataContext;
+  };
+  mapPointCoordinates: {
+    data: Array<MapPoint>;
+    context?: DataContext;
+  };
+}
+
+export const emptyChartStore = {
+  geoJson: {
+    data: null,
+  },
+  defaultStat: {
+    data: null,
+  },
+  allStat: {
+    data: null,
+  },
+  denomStat: {
+    data: null,
+  },
+  breadcrumbStat: {
+    data: null,
+  },
+  breadcrumbDenomStat: {
+    data: null,
+  },
+  mapPointStat: {
+    data: null,
+  },
+  mapPointCoordinates: {
+    data: null,
+  },
+};
+
+/**
+ * Check whether data is ready to use.
+ *
+ * Data is ready if the context in ChartStore matches the current context.
+ *
+ * @param ctx The data context from chart store.
+ * @param type The type of the data.
+ * @param placeInfo The current placeInfo from the context.
+ * @param statVar The current statVar from the context.
+ * @returns
+ */
+export const isDataReady = (
+  dataWrapper: { data: any; context?: DataContext },
+  type: ChartDataType,
+  placeInfo: PlaceInfo,
+  statVar: StatVar
+): boolean => {
+  if (_.isEmpty(dataWrapper)) {
+    return false;
+  }
+  const ctx = dataWrapper.context;
+  if (_.isEmpty(ctx)) {
+    return false;
+  }
+  switch (type) {
+    case ChartDataType.geoJson:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
+        placeInfo.enclosedPlaceType === ctx.placeInfo.enclosedPlaceType
+      );
+    case ChartDataType.defaultStat:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
+        placeInfo.enclosedPlaceType === ctx.placeInfo.enclosedPlaceType &&
+        statVar.dcid === ctx.statVar.dcid &&
+        statVar.date === ctx.statVar.date
+      );
+    case ChartDataType.allStat:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
+        placeInfo.enclosedPlaceType === ctx.placeInfo.enclosedPlaceType &&
+        statVar.dcid === ctx.statVar.dcid &&
+        statVar.date === ctx.statVar.date
+      );
+    case ChartDataType.denomStat:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
+        placeInfo.enclosedPlaceType === ctx.placeInfo.enclosedPlaceType &&
+        statVar.denom === ctx.statVar.denom
+      );
+    case ChartDataType.breadcrumbStat:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
+        statVar.dcid === ctx.statVar.dcid &&
+        statVar.date === ctx.statVar.date
+      );
+    case ChartDataType.breadcrumbDenomStat:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
+        statVar.denom === ctx.statVar.denom
+      );
+    case ChartDataType.mapPointStat:
+      return (
+        placeInfo.enclosingPlace.dcid === ctx.placeInfo.enclosingPlace.dcid &&
```

Comment:
need to also check that placeInfo.mapPointPlaceType is the same

Reply:
Done